src/app/Shared/Services/Settings.service.tsx

```diff
@@ -79,6 +86,49 @@ export class SettingsService {
     return defaultUnits;
   }
 
+  notificationsEnabled(): Map<NotificationCategory, boolean> {
+    const raw = window.localStorage.getItem(StorageKeys.NotificationsEnabled);
+    if (!!raw) {
+      try {
+        const map = JSON.parse(raw);
+        if (typeof map === 'object') {
+          const obj = new Map(Array.from(Object.entries(map)));
```

Comment:
Fair enough, that works too, although accessing the elements of the inner pairs looks even funnier this way - but it does work.

Reply:
Yeah, it does look funnier. I'll leave it up to you whether you'd like to revert it to the previous way or not. My question was more about wondering if there were some Typescript typecasting intricacies that I wasn't aware of.

Reply:
Yea, I tried a few different ways to iterate over this but nothing came out as nice as I wanted. We get foiled by the need to deserialize it from JSON, basically. This gives us back a plain Object, but I wanted to cast/convert it into a Map so that the deserialization side was symmetric with the serialization side, and so that the Map object could be dealt with an methods like `has`, `set`, `get` would be available - it's the right abstract data type for this purpose.